### PR TITLE
Added comments to Source files.

### PR DESCRIPTION
**Pull request by xiane**
Added comments to source files.
 Target files : lpae._, mm._, and smp.h
Fixed wrong defined mask.
- smp.h : cpu id of mpidr
-  lpae.h : Level 1 Output address mask

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

File: hypervisor/hardware/arm32ve/libhw/lpae.c [hunk 1]
```diff
@@ -3,13 +3,27 @@
...
+ * The address mask of each level descriptors.
+ * - TTBL_L1_OUTADDR_MASK[39:30] Level1 Block address mask.
+ * - TTBL_L2_OUTADDR_MASK[39:21] Level2 Block Address mask.
+ * - TTBL_L3_OUTADDR_MASK[39:12] Page address mask.
+ *
+ * - TTBL_L1_TABADDR_MASK[39:12] Level2 table descriptor address mask.
+ * - TTBL_L2_TABADDR_MASK]30:12] Level3 table descriptor address mask.
+ * @{
+ */
+#define TTBL_L1_OUTADDR_MASK    0x000000FFC0000000ULL
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.c:18]: What does it mean? and What is the root cause before your change?

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 2]
```diff
@@ -2,21 +2,233 @@
...
 #define LPAE_PAGE_SIZE      (1<<LPAE_PAGE_SHIFT)
 #define LPAE_PAGE_MASK      (0xFFF)
+/**
+ * @}
+ */
 
+/**
+ * \defgroup LPAE_BLOCK_FEATURES
+ *
+ * This features are used to configure the Bloack addres of 2MB size.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:199]: s/Bloack/block

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 3]
```diff
@@ -24,103 +236,176 @@
...
+ *
+ * This descriptor has almost the same layout to lpae_pt.
+ * But it is stage-2 translation descriptor.
+ * - \ref Long_descriptor "Details"
+ *
+ * - Description
+ *
+ *   - These are used in all kinds of entry.
+ *     - valid - Valid mapping
+ *     - table - == 1 Table Entry, == 0 block or page Entry.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:328]: table - == 1 Table Entry, == 0 block or page Entry.
If the table is zero then it means table entry?

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 4]
```diff
@@ -128,34 +413,221 @@ struct lpae_walk {
...
+    LPAED_STAGE2_MEMATTR_NORMAL_OWT = 0x8,  /**< Outer Write-through */
+    LPAED_STAGE2_MEMATTR_NORMAL_OWB = 0xC,  /**< Outer Write-back */
+    LPAED_STAGE2_MEMATTR_NORMAL_INC = 0x1,  /**< Inner Non-cacheable */
+    LPAED_STAGE2_MEMATTR_NORMAL_IWT = 0x2,  /**< Inner Write-through */
+    LPAED_STAGE2_MEMATTR_NORMAL_IWB = 0x3,  /**< Inner Write-back */
 };
 
+/**
+ * @brief Level 1 block, 1GB, entry in LPAE descriptor format
+ * for the given physical addres.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:434]: s/addres/address

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 5]
```diff
@@ -128,34 +413,221 @@ struct lpae_walk {
...
+/**
+ * @brief Configure the stage-2 level 2 table entry's valid & table bit
+ * and set the base address.
+ *
+ * Configure the descriptor's valid bit and table bit.
+ * And mapping physical address of the level 3 table.
+ *
+ * - State configuration
+ *   - valid = valid ? 1: 0, table = valid ? 1 : 0
+ *   - bass address = baddr
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:599]: s/bass/base

File: hypervisor/mm.c [hunk 6]
```diff
@@ -148,19 +219,32 @@ void hmm_heap_init(void)
  * PL2 Stage1 Translation
  * VA32 -> PA
  */
-
+/**
+ * @brief Initializes the host monitor memory management.
```
Comment (Wonseok) [inline hypervisor/mm.c:223]: what is the host monitor?

Comment (Wonseok) [inline hypervisor/mm.c:223]: Did you mean khypervisor?

File: hypervisor/mm.c [hunk 7]
```diff
@@ -390,6 +481,18 @@ void hmm_flushTLB(void)
     asm volatile("isb");
 }
 
+/**
+ * @brief Returns the level 3 table entry.
+ *
+ * Returns the level3 translation table descriptor by matching
```
Comment (Wonseok) [inline hypervisor/mm.c:487]: s/level3/level 3

File: hypervisor/mm.c [hunk 8]
```diff
@@ -390,6 +481,18 @@ void hmm_flushTLB(void)
...
+ * The virtual address and number of pages.
```
Comment (Wonseok) [inline hypervisor/mm.c:488]: combine L488-L490 into 2 lines.

File: hypervisor/mm.c [hunk 9]
```diff
@@ -475,6 +584,17 @@ void hmm_free(void *ap)
     freep = p;
 }
 
+/**
+ * @brief Obtains storage from the heap memory.
+ *
+ * Obtains storage from hyp mode heap memory.
```
Comment (Wonseok) [inline hypervisor/mm.c:590]: Please make it concatenated paragraph.

File: hypervisor/mm.c [hunk 10]
```diff
@@ -475,6 +584,17 @@ void hmm_free(void *ap)
...
+ * If wanted size of block is not bigger then NALLOC value,
+ * This function obtains minimally NALLOC bytes(1024 Bytes).
+ * After obtains storage, makes free this area by call free function.
+ *
+ * @param nu Number of units, allocate size.
+ * @return Pointer of free list block header & memory area.
```
Comment (Wonseok) [inline hypervisor/mm.c:596]: How big is the unit? 4KB? or ?

**Comment by Wonseok** [comment]
Overall looks good to me. A minor comments left. Please take a look.
Your comments still is too long. You don't need to write all the details in comment.

**Pull request closed without merging**